src/backend/distributed/transaction/transaction_management.c

```diff
@@ -460,6 +461,12 @@ ResetGlobalVariables()
 	CoordinatedTransactionUses2PC = false;
 	TransactionModifiedNodeMetadata = false;
 
+	/*
+	 * Make sure that we give the shared connections back to the shared
+	 * pool if any. This operation is a no-op if the reserved connections
+	 * are already given away.
+	 */
+	DeallocateAllReservedConnections();
```

Comment:
this is in ResetGlobalVariables, but feels a bit more advanced than resetting a global variable